### PR TITLE
Add authz flow diagram for social agent sharing access

### PR DESCRIPTION
**Pull request by woutermont**
This diagram works out the third of the authorization flows [section 7.1](https://solid.github.io/data-interoperability-panel/specification/#authorization-flows): Social Agent Shares Access. Feedback is welcome!

### CODE REVIEW

**Comment by justinwb** [comment]
Overall +1 on including diagrams for these specific flows. A few bits of feedback:

1. Needs slight adjustment to align with change in terminology for Access Consent -> Access Authorization
2. Per discussion on 4/19 - we need to break out a bit more what access needs and context should be provided to the authorization agent

**Comment by elf-pavlik** [comment]
Now Github has full support for Mermaid https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

One can select *View file* and see the [rendered diagram](https://github.com/solid/data-interoperability-panel/blob/2f142a6c238698dabb4b368152776c2a0ce8a918/proposals/specification/diagrams/social-agent-shares-access.seq.mmd]

Apologies @woutermont I hope we can get back to this PR next week ⏲️

File: proposals/specification/diagrams/social-agent-shares-access.seq.mmd [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    Note over proj: 6. 👩 Alice decides to share a project with Bob
+    Note over proj: 7. Projectron generates Access Needs for Bob
+    
+    proj-->>aa_alice: 8. Redirects to AA with Bob's Access Needs
+    
+    Note over id_proj, aa_alice: 9. AA discovers Projectron's redirect uri
+    Note over aa_alice: 10. Presents Alice with Access Needs
+    Note over aa_alice: 11. 👩 Alice reviews Access Needs
+    Note over aa_alice: 12. 👩 Alice chooses scope of access for Bob
+    Note over aa_alice, pod_alice: 14. Record Access Consent
```
Comment (elf-pavlik) [inline proposals/specification/diagrams/social-agent-shares-access.seq.mmd:28]:
```suggestion
    Note over aa_alice, pod_alice: 14. Record Access Authorization
```

**Review by elf-pavlik — APPROVED** [review]
I think this is a good initial diagram. As everything it will most likely improve further over some iterations.

**Review by justinwb — APPROVED** [review]
✅

**Pull request merged**